src/mbgl/storage/request_holder.hpp

```diff
@@ -0,0 +1,26 @@
+#ifndef MBGL_STORAGE_REQUEST_HOLDER
+#define MBGL_STORAGE_REQUEST_HOLDER
+
+#include <memory>
+
+namespace mbgl {
+
+class Request;
+
+class RequestHolder {
```

Comment:
I implemented this at first, but backtracked because the pull request grew too big due to too many changes necessary for the unit tests. I'll try again.